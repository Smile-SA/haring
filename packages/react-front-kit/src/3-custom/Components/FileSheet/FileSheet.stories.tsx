import type { Meta, StoryObj } from '@storybook/react';

import { FileSheet as Cmp } from './FileSheet';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/FileSheet',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const FileSheet: IStory = {
  args: {
    dropZone: false,
    motifVisible: true,
    title: <h1>Jean-Michel DUPONT</h1>,
  },
};
