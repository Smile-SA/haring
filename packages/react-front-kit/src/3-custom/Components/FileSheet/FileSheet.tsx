'use client';

import { Box, BoxProps, MantineProvider } from '@mantine/core';
import { createStyles } from '@mantine/styles';
import type { ReactElement, ReactNode } from 'react';
import Motif from './Motif';

interface IFileSheetProps extends BoxProps {
  children?: ReactNode;
  title?: ReactElement;
  cards?: {title?: 'string', image?: ReactElement, action?: () => {}}[],
  content?: ReactElement,
  dropZone: boolean,
  dropZoneContent?: ReactElement,
  motif?: ReactElement,
}


const useStyles = createStyles((theme) => ({
  fileSheet: {
    backgroundColor: theme.black,
    padding: '32px 42px', 
    borderRadius: '16px',
    width: '100%',
    minHeight: '219px',
    position: 'relative',
    overflow: 'hidden',
  },
  title: {
    'h1, h2, h3, h4 h5, p': {
      fontSize: '26px',
      color: theme.white,
      fontWeight: 700,
      marginBottom: '24px',
    }
  },
  leftContainer: {
    maxWidth: '342px',
  },
  rightContainer: {

  },
  cards: {

  },
  card: {

  },
  content: {

  },
  motif: {
    position: 'absolute',
    top: 0,
    left: 0,
  }
}));


export function FileSheet(props: IFileSheetProps): ReactElement {
  const { children, title, cards,
  content,
  dropZone,
  dropZoneContent, motif, ...BoxProps } = props;
  const { classes } = useStyles();
  return (
    <MantineProvider theme={{colorScheme: 'dark'}}>
    <Box className={classes.fileSheet} {...BoxProps}>
      {!motif == null &&     
        <div className={classes.motif}>
          {motif ? motif : <Motif/>}
        </div>
      }
      <div className={classes.leftContainer}>
        { title && 
          <div className={classes.title}>
            {title}
          </div>
        }
        <div className="cards">
          { cards && cards.map(item => 
            <div>
              {
                item.image ? item.image : ""
              }
              <span>
                {item.title && item.title}
              </span>
            </div>
          )}
        </div>
        {content &&  
          <div className={classes.content}>
            {content}
          </div>
        }
      </div>
      {dropZone &&   
        <div className={classes.rightContainer}>

        </div>
      }
    </Box>
    </MantineProvider>
  );
}
