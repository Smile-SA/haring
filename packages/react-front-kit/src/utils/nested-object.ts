export type INestedObjectWithDepth<O extends object> = O & {
  children?: INestedObjectWithDepth<O>[];
  depth?: number;
};

export function flattenNestedObjects<O extends object>(
  arr: INestedObjectWithDepth<O>[],
): INestedObjectWithDepth<O>[] {
  let children: INestedObjectWithDepth<O>[] = [];
  const flattened = arr.map((m) => {
    if (m.children && m.children.length) {
      children = [...children, ...m.children];
    }
    return m;
  });
  return flattened.concat(
    children.length ? flattenNestedObjects(children) : children,
  );
}

export function assignDepth<O extends object>(
  arr?: INestedObjectWithDepth<O>[],
  depth = 0,
): INestedObjectWithDepth<O>[] {
  (arr ?? []).forEach((o) => {
    o.depth = depth;
    assignDepth(o.children, depth + 1);
  });
  return arr ?? [];
}
