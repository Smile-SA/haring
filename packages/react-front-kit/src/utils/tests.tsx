import type {
  Queries,
  RenderOptions,
  RenderResult,
  queries,
} from '@testing-library/react';
import type { ReactElement, ReactNode } from 'react';

import { render } from '@testing-library/react';

import { Provider } from '../3-custom/Provider/Provider';

export function renderWithProviders<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container
>(
  ui: ReactElement,
  renderOptions: RenderOptions<Q, Container, BaseElement> = {}
): RenderResult<Q, Container, BaseElement> {
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return <Provider>{children}</Provider>;
  }
  return render<Q, Container, BaseElement>(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}
