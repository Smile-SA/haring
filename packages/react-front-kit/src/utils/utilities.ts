export function isNotNullNorEmpty<S>(
  value: Record<string, never> | S,
): value is Exclude<S, null | undefined> {
  return (
    value !== null &&
    value !== undefined &&
    (typeof value === 'object' ? Object.keys(value).length !== 0 : true)
  );
}

export function isCallback<T, U>(maybeFunc: T | U): maybeFunc is T {
  return typeof maybeFunc === 'function';
}
