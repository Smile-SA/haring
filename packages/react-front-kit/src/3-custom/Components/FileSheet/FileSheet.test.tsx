import { renderWithProviders } from '../../../utils/tests';

import { FileSheet } from './FileSheet';

describe('FileSheet', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(<FileSheet dropZone={false} />);
    expect(container).toMatchSnapshot();
  });
});
