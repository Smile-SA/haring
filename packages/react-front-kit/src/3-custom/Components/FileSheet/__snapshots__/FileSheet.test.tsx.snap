// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileSheet matches snapshot 1`] = `
<div>
  <div
    class="mantine-1c216y3"
  >
    <div
      class="mantine-quehkv"
    >
      <div
        class="cards"
      />
    </div>
  </div>
</div>
`;
